@copilot some tests are failing: 
FAILED tests/test_private_ai.py::test_delete_private_ai_key - ValueError: Inv...
FAILED tests/test_private_ai.py::test_delete_private_ai_key_with_only_vector_db
======= 2 failed, 729 passed, 2 skipped, 6 warnings in 357.03s (0:05:57) =======
make: *** [Makefile:69: backend-test-cov] Error 1
Error: Process completed with exit code 2. 

Fix failing tests.

--- a/app/db/postgres.py
+++ b/app/db/postgres.py
@@ -106,6 +106,10 @@ class PostgresManager:
     async def delete_database(
         self, database_name: str, database_username: str | None = None
     ):
+        _validate_identifier(database_name, "database name")
+        if database_username:
+            _validate_identifier(database_username, "database username")
+
         conn = await asyncpg.connect(
             host=self.host,
             port=self.port,
@@ -114,7 +118,6 @@ class PostgresManager:
         )
 
         try:
-            _validate_identifier(database_name, "database name")
             # Terminate all connections to the database (parameterized to prevent injection)
             await conn.execute(
                 """
@@ -126,7 +129,6 @@ class PostgresManager:
             )
             await conn.execute(f"DROP DATABASE IF EXISTS {database_name}")
             if database_username:
-                _validate_identifier(database_username, "database username")
                 await conn.execute(
                     """
                     SELECT pg_terminate_backend(pg_stat_activity.pid)

--- a/tests/test_private_ai.py
+++ b/tests/test_private_ai.py
@@ -99,10 +99,10 @@ def test_delete_private_ai_key(
 
     # Create a test private AI key
     test_key = DBPrivateAIKey(
-        database_name="test-db-delete",
+        database_name="test_db_delete",
         name="Test Key to Delete",
         database_host="test-host",
-        database_username="test-user",
+        database_username="test_user",
         database_password="test-pass",
         litellm_token="test-token-delete",
         litellm_api_url="https://test-litellm.com",
@@ -1940,10 +1940,10 @@ def test_delete_private_ai_key_with_only_vector_db(
     """Test deleting a private AI key that only has a vector database (no LLM token)"""
     # Create a test key with only vector DB
     test_key = DBPrivateAIKey(
-        database_name="test-db-vector-only",
+        database_name="test_db_vector_only",
         name="Test Vector DB Only",
         database_host="test-host",
-        database_username="test-user",
+        database_username="test_user",
         database_password="test-pass",
         litellm_token=None,  # No LLM token
         litellm_api_url=None,  # No LLM API URL

